packages/cozy-doctypes/README.md

```diff
@@ -6,7 +6,121 @@ This repository is meant to bring together methods and functions for doctypes th
+It can be used as a base for other document types (see [BankTransaction](https://github.com/cozy/cozy-libs/blob/master/packages/cozy-doctypes/src/banking/BankTransaction.js) for example).
 
 ⚠️ The Document API does not work well in contexts where you need to change the client in flight. For this, prefer to use
 directly the `cozy-client` APIs.
+
+### How to use it
+
+Let's say we have a `io.cozy.simpsons` doctype, because we are developing a konnector that retrieves some Simpsons data. We create a class `Simpson` that inherits from `Document` :
+
+```js
+const { Document } = require('cozy-doctypes')
+
+class Simpson extends Document {}
+```
+
+The first thing we are going to do is specify the doctype:
+
+```js
+const { Document } = require('cozy-doctypes')
+
+class Simpson extends Document {}
+Simpson.doctype = 'io.cozy.simpsons'
+```
+
+Then we attach a `cozy-client-js` instance to it. We are going to use [cozy-konnector-libs](https://github.com/konnectors/libs/tree/master/packages/cozy-konnector-libs) to get this instance. Take a look at the [konnectors documentation](https://docs.cozy.io/en/tutorials/konnector/) to learn more about that.
+
+```
+const { Document } = require('cozy-doctypes')
+const { cozyClient } = require('cozy-konnector-libs')
+
+class Simpson extends Document {}
+Simpson.doctype = 'io.cozy.simpsons'
+
+Simpson.registerClient(cozyClient)
+```
+
+With this, we can already use our class. For example, we can create a new document:
+
+```js
+Simpson.create({
+  name: 'Homer'
+})
+```
+
+Or get some documents given their ids:
+
+```js
+const documents = await Simpson.getAll(['id1', 'id2', 'id3']
+```
+
+Take a look at [the Document class](https://github.com/cozy/cozy-libs/blob/master/packages/cozy-doctypes/src/Document.js) to see all the methods it implements.
+
+### Special class properties
+
+There are some properties that can be given to `Document` that have a special meaning.
+
+#### `idAttributes`
+
+`idAttributes` is an array of attributes that are seen as ids. When `Document` knows that, it becomes able to determine if a given document must be created or updated. This is used in the `createOrUpdate` method. Let's say that we want the `name` of our `Simpson`s to be an id:
+
+```js
+Simpson.idAttributes = ['name']
+```
+
+Now, if we do the following:
+
+```js
+Simpson.createOrUpdate({ name: 'Homer', description: 'Likes donuts' })
+```
+
+Since we said that `name` is an id attribute, the document we previously created with the name « Homer » will be updated. But if we do:
+
+```js
+Simpson.createOrUpdate({ name: 'Marge', description: "Homer's wife" })
+```
+
+This document will be created, because no document exists with this `name`.
+
+#### `checkedAttributes`
+
+`checkedAttributes` is also an array of attributes. But it is used to determine if a document should be updated or not. Let's say that we want a document to be updated only if its `description` changed:
```

Comment:
When skimming the documentation, it is possible, that you skip the previous section. When reading this you might have forgotten the previous one, hence I would skip the "also".